specifyweb/frontend/js_src/lib/components/DataModel/collectionApi.ts

```diff
@@ -206,7 +229,7 @@ export const IndependentCollection = LazyCollection.extend({
       'remove',
       function (resource: SpecifyResource<AnySchema>) {
         if (!resource.isNew()) {
-          (this.removed as Set<string>).add(resource.url());
+          (this.removed as ReadonlySet<string>).add(resource.url());
```

Comment:
is this valid in typescript?
you are calling `.add()` on a ReadonlySet?
isn't readonly supposed to prevent changes?

Reply:
Ah yeah, interesting. 
It's not valid TS. 

We have a ESlint rule to prevent directly using a Set (immutable types are preferred). 
ESlint autofixed the instances where I casted the type as Set and changed it to ReadonlySet, but still kept the mutable methods (which don't exist on ReadonlySet).

Reply:
ah, haha makes sense
also, because of ts-nocheck, eslint may be starving of some type information and thus behaving odly